From: Dan Williams <dan.j.williams@intel.com>
Date: Wed, 6 Jan 2016 12:19:22 -0800
Subject: badblocks: rename badblocks_free to badblocks_exit
Git-commit: d3b407fb3f782bd915db64e266010ea30a2d381e
Patch-mainline: v4.5-rc1
References: FATE#319858

For symmetry with badblocks_init() make it clear that this path only
destroys incremental allocations of a badblocks instance, and does not
free the badblocks instance itself.

Signed-off-by: Dan Williams <dan.j.williams@intel.com>
Acked-by: Johannes Thumshirn <jthumshirn@suse.com>
---
 block/badblocks.c         |    6 +++---
 block/genhd.c             |    2 +-
 drivers/md/md.c           |    2 +-
 include/linux/badblocks.h |    2 +-
 4 files changed, 6 insertions(+), 6 deletions(-)

--- a/block/badblocks.c
+++ b/block/badblocks.c
@@ -550,12 +550,12 @@ int badblocks_init(struct badblocks *bb,
 EXPORT_SYMBOL_GPL(badblocks_init);
 
 /**
- * badblocks_free() - free the badblocks structure
+ * badblocks_exit() - free the badblocks structure
  * @bb:		the badblocks structure that holds all badblock information
  */
-void badblocks_free(struct badblocks *bb)
+void badblocks_exit(struct badblocks *bb)
 {
 	kfree(bb->page);
 	bb->page = NULL;
 }
-EXPORT_SYMBOL_GPL(badblocks_free);
+EXPORT_SYMBOL_GPL(badblocks_exit);
--- a/block/genhd.c
+++ b/block/genhd.c
@@ -671,7 +671,7 @@ void del_gendisk(struct gendisk *disk)
 	blk_unregister_region(disk_devt(disk), disk->minors);
 
 	if (disk->bb) {
-		badblocks_free(disk->bb);
+		badblocks_exit(disk->bb);
 		kfree(disk->bb);
 	}
 
--- a/drivers/md/md.c
+++ b/drivers/md/md.c
@@ -711,7 +711,7 @@ void md_rdev_clear(struct md_rdev *rdev)
 		put_page(rdev->bb_page);
 		rdev->bb_page = NULL;
 	}
-	badblocks_free(&rdev->badblocks);
+	badblocks_exit(&rdev->badblocks);
 }
 EXPORT_SYMBOL_GPL(md_rdev_clear);
 
--- a/include/linux/badblocks.h
+++ b/include/linux/badblocks.h
@@ -48,6 +48,6 @@ ssize_t badblocks_show(struct badblocks
 ssize_t badblocks_store(struct badblocks *bb, const char *page, size_t len,
 			int unack);
 int badblocks_init(struct badblocks *bb, int enable);
-void badblocks_free(struct badblocks *bb);
+void badblocks_exit(struct badblocks *bb);
 
 #endif
